auth0/v3/authentication/token_verifier.py

```diff
@@ -340,3 +339,130 @@ def _verify_payload(self, payload, nonce=None, max_age=None, organization=None):
+                'found "{}"'.format(self.iss, payload['iss']))
+
+        # Subject
+        if 'sub' not in payload or not isinstance(payload['sub'], (str, ustr)):
+            raise TokenValidationError('Subject (sub) claim must be a string present in the access token')
+
+        # Audience
+        if 'aud' not in payload or not (isinstance(payload['aud'], (str, ustr)) or isinstance(payload['aud'], list)):
+            raise TokenValidationError(
+                'Audience (aud) claim must be a string or array of strings present in the access token')
+
+        if isinstance(payload['aud'], list) and self.aud not in payload['aud']:
+            payload_audiences = ", ".join(payload['aud'])
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" but was '
+                'not one of "{}"'.format(self.aud, payload_audiences))
+        elif isinstance(payload['aud'], (str, ustr)) and payload['aud'] != self.aud:
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" '
+                'but found "{}"'.format(self.aud, payload['aud']))
+
+        # --Time validation (epoch)--
+        now = self._clock or time.time()
+        leeway = self.leeway
+
+        # Expires at
+        if 'exp' not in payload or not isinstance(payload['exp'], int):
+            raise TokenValidationError('Expiration Time (exp) claim must be a number present in the access token')
+
+        exp_time = payload['exp'] + leeway
+        if now > exp_time:
+            raise TokenValidationError(
+                'Expiration Time (exp) claim error in the access token; current time ({}) is '
+                'after expiration time ({})'.format(now, exp_time))
+
+        # Issued at
+        if 'iat' not in payload or not isinstance(payload['iat'], int):
+            raise TokenValidationError('Issued At (iat) claim must be a number present in the access token')
+
+        # Scopes
+        if scopes:
+            if 'scopes' not in payload or not isinstance(payload['scopes'], str):
+                raise TokenValidationError('No scopes claim found in token.')
+
+            for scope in scopes.split(' '):
+                if scope not in payload['scopes'].split(' '):
+                    msg = "You don't have access to this resource"
+                    raise Auth0Error(status_code=401, error_code=401, message=msg)
+
+        # Permissions
+        if permissions:
+            if 'permissions' not in payload or not isinstance(payload['permissions'], list):
+                raise TokenValidationError('No permissions claim found in token.')
+
+            for permission in permissions:
+                if permission not in payload['permissions']:
+                    msg = "You don't have access to this resource"
+                    raise Auth0Error(status_code=403, error_code=403, message=msg)
```

Comment:
I'm not sure this is the correct error status code or error code and message for this case

Reply:
`403` Forbidden is correct when the user does not have the correct permissions